packages/server/graphql/mutations/setPokerSpectate.ts

```diff
@@ -47,7 +69,19 @@ const setPokerSpectate = {
         .get(teamMemberId)
         .update({isSpectatingPoker: isSpectating, updatedAt: now})
     }).run()
-    const data = {meetingId, userId: viewerId}
+
+    // mutate the dataLoader cache
+    meetingMember.isSpectating = isSpectating
+    stages.forEach(async (stage) => {
+      const {id: stageId, scores} = stage
+      const viewerVotedInStage = scores.find(({userId}) => userId === viewerId)
+      if (viewerVotedInStage) {
+        const newStageScores = scores.filter(({userId}) => userId !== viewerId)
+        stage.scores = newStageScores
+        await removeVoteForUserId(viewerId, stageId, meetingId)
+      }
+    })
+    const data = {meetingId, userId: viewerId, stages, teamId, phaseType: 'ESTIMATE'}
```

Comment:
+1 I think we could optimize this slightly to match your description of this field, i.e. maintain a second array `updatedStages` and only send this here instead of all stages.